main.py

```diff
@@ -148,6 +148,17 @@ async def main():
 
     urls = [video["titleUrl"] for video in kept if "titleUrl" in video]
 
+    async with aiofiles.open("history.log", "a+") as h_file:
+        await h_file.seek(0)
+        processed: list[str] = [line.rstrip() for line in await h_file.readlines()]
+
+    async with aiofiles.open("failed.log", "a+") as f_file:
+        await f_file.seek(0)
+        failed: list[str] = [line.rstrip() for line in await f_file.readlines()]
+
+        urls = list(set(urls) - set(processed + failed))
+        logger.info(f"{len(processed)} URL has already been processed. Skipping...")
+
```

Comment:
Logging to log files shouldn't be necessary. The script should ideally not fail at all and if it does, it should be fixed.

Reply:
In my case some videos are made private, deleted etc. so made a log file to exclude in second run and keep track those if necessary for someone.

Reply:
How about just logging it in the CLI output? The output can be refactored to look like this:

Restored 1/100 videos.

Failed to restore (10 videos):
- link1
- link2

Reply:
It will be not excluded in second run. That's the reason history.log is saved

Reply:
There should not be a need to run this script twice. If so, the need to should be fixed instead

Reply:
If the history is too long like 20,000 then it takes too much time. So, user can run multiple times when free.